### PR TITLE
Fix/duplicate events on onboarding

### PR DESCRIPTION
**Pull request by femyeda**
-   dep: add types for lodash.debounce
-   fix issue causing many duplicate event types
    add loader after user actions
    debounce actions to prevent multiple clicks

- fix user bio not persiting to db
-  add rest api for fetch all event types for user
    add query for fetching all event types for user

### CODE REVIEW

**Comment by femyeda** [comment]
Overview of Changes
[https://www.loom.com/share/5e03f25d453d43fdb71f7904801cec08](https://www.loom.com/share/5e03f25d453d43fdb71f7904801cec08)

Follow Up Video about Schedule
[https://www.loom.com/share/06bc3c8ee3d24a44b40d9af4d5048e89](https://www.loom.com/share/06bc3c8ee3d24a44b40d9af4d5048e89)

**Review by zomars — APPROVED** [review]
Looking good 👍🏽

**Review by pumfleet — APPROVED** [review]
Looks good as far as I can see. Great work :)

File: pages/api/event-type/index.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            length: true,
+            schedulingType: true,
+            slug: true,
+            hidden: true,
+          },
+        },
+      },
+    });
+
+    return res.status(200).json({ message: "Events.", data: user.eventTypes });
```
Comment (emrysal) [inline pages/api/event-type/index.ts:42]: Suggestion, as it is consistent
```javascript
.json({ events: user.eventTypes })
```

File: pages/getting-started.tsx [hunk 2]
```diff
@@ -30,6 +30,9 @@ import classnames from "classnames";
 import { ArrowRightIcon } from "@heroicons/react/outline";
 import { getSession } from "@lib/auth";
 import Button from "@components/ui/Button";
+import debounce from "lodash.debounce";
+import Loader from "@components/Loader";
+import getEventTypes from "../lib/queries/event-types/get-event-types";
```
Comment (emrysal) [inline pages/getting-started.tsx:35]: @lib/

**Review by emrysal — APPROVED** [review]
Would love @KATT to have a look as well given he knows react-query much more than I do - may be able to leverage it more.

**Review by zomars — APPROVED** [review]

File: pages/getting-started.tsx (hunk 2 above)
Comment (zomars) [reply pages/getting-started.tsx:35]:
```suggestion
import getEventTypes from "@lib/queries/event-types/get-event-types";
```

File: pages/api/event-type/index.ts (hunk 1 above)
Comment (femyeda) [reply pages/api/event-type/index.ts:42]: Returning {message: "...", data: {...}} is a common standard as your fetcher doesn't need to know the exact shape of the response for generalization only the expected type of data, if someone wants to change it feel free to do so.

**Pull request merged**